airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/component/TableSchemaEvolutionSuite.kt

```diff
@@ -0,0 +1,422 @@
+import io.airbyte.cdk.load.data.IntegerType
+import io.airbyte.cdk.load.data.IntegerValue
+import io.airbyte.cdk.load.data.ObjectType
+import io.airbyte.cdk.load.data.ObjectValue
+import io.airbyte.cdk.load.data.StringType
+import io.airbyte.cdk.load.data.StringValue
+import io.airbyte.cdk.load.data.TimestampWithTimezoneValue
+import io.airbyte.cdk.load.message.Meta
+import io.airbyte.cdk.load.message.Meta.Companion.COLUMN_NAME_AB_EXTRACTED_AT
+import io.airbyte.cdk.load.message.Meta.Companion.COLUMN_NAME_AB_GENERATION_ID
+import io.airbyte.cdk.load.message.Meta.Companion.COLUMN_NAME_AB_META
+import io.airbyte.cdk.load.message.Meta.Companion.COLUMN_NAME_AB_RAW_ID
+import io.airbyte.cdk.load.table.ColumnNameMapping
+import io.airbyte.cdk.load.table.TableName
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest
+import kotlin.test.assertEquals
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.assertAll
+
+@MicronautTest(environments = ["component"])
+interface TableSchemaEvolutionSuite {
+    val client: TableSchemaEvolutionClient
+    val airbyteMetaColumnMapping: Map<String, String>
+        get() = Meta.COLUMN_NAMES.associateWith { it }
+
+    val opsClient: TableOperationsClient
+    val testClient: TestTableOperationsClient
+
+    private val harness: TableOperationsTestHarness
+        get() = TableOperationsTestHarness(opsClient, testClient, airbyteMetaColumnMapping)
+
+    /**
+     * Test that the connector can correctly discover all of its own data types. This test creates a
+     * table with a column for each data type, then tries to discover its schema.
+     *
+     * @param expectedDiscoveredSchema The schema to expect. This should be the same schema as
+     * [computeSchema].
+     */
+    fun discover(expectedDiscoveredSchema: TableSchema) = runTest {
+        val testNamespace = Fixtures.generateTestNamespace("namespace-test")
+        val testTable = Fixtures.generateTestTableName("table-test-table", testNamespace)
+        val stream =
+            Fixtures.createAppendStream(
+                namespace = testTable.namespace,
+                name = testTable.name,
+                schema = Fixtures.ALL_TYPES_SCHEMA,
+            )
+
+        opsClient.createNamespace(testNamespace)
+        opsClient.createTable(
+            tableName = testTable,
+            columnNameMapping = Fixtures.ALL_TYPES_MAPPING,
+            stream = stream,
+            replace = false,
+        )
+
+        val discoveredSchema = client.discoverSchema(testTable)
+        assertEquals(expectedDiscoveredSchema, discoveredSchema)
+    }
+
+    /**
+     * Test that the connector can correctly compute a schema for a stream containing all data
+     * types.
+     *
+     * @param expectedComputedSchema The schema to expect. This should be the same schema as
+     * [discover].
+     */
+    fun computeSchema(
```

Comment:
this test maybe eventually belongs in the type/name munging suite :shrug: it's here for now b/c computeSchema is in the schema evolution client though